### ISSUE TITLE
Takes too long before starting the test execution

### ISSUE THREAD

**Issue by indrajitbnikam** [cypress-io/cypress#3692]
### Current behavior:

Cypress is taking **too much time to initialize the cypress** before starting the test actual execution. 
Actual execution of test takes only **~8-10 seconds**, but when I start test using CLI i.e. using 
```sh
cypress run -- --spec "cypress/integration/specname" --browser chrome
```
it will take around **30 seconds** before it even starts the chrome.

Currently I'm working on windows 10, I have seen many tutorials and they seemed fast on mac OS. Is it in any way OS dependent when it comes to performance? especially, when you are running test through CLI.

### Desired behavior:
It should only take few seconds if not zero, before it starts test execution

### Steps to reproduce:

1. First of all I have **configured** my project in order to write the **test cases in typescript**.
2. My Spec file,
```js
describe('My First Cypress Test', () => {
    it('Not much', () => {
        cy.visit('https://www.google.co.in/');
    });
});
```

3. Now you need to run the code using command,
```sh
cypress run -- --spec "cypress/integration/specname" --browser chrome
```
or
```sh
npm run cypress:run -- --spec "cypress/integration/specname" --browser chrome
```

### Versions
Windows: 10 Enterprise
this is my package.json
```json
{
  "name": "cypress-spike",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "cypress:run": "./node_modules/.bin/cypress run",
    "cypress:open": "./node_modules/.bin/cypress open",
    "test-junit": "cypress run --reporter junit --reporter-options 'mochaFile=junit-results/my-test-output.xml'",
    "test-multiple": "cypress run --reporter mocha-multi-reporters --reporter-options configFile=config.json",
    "delete:reports": "rm cypress/results/* || true",
    "prereport": "npm run delete:reports"
  },
  "dependencies": {},
  "devDependencies": {
    "@cypress/webpack-preprocessor": "^4.0.3",
    "@types/node": "^11.9.4",
    "cypress": "^3.1.5",
    "cypress-xpath": "^1.3.0",
    "mocha": "^6.0.1",
    "mocha-junit-reporter": "^1.18.0",
    "mocha-multi-reporters": "^1.1.7",
    "mochawesome": "^3.1.1",
    "ts-loader": "^5.3.3",
    "typescript": "^3.3.3",
    "webpack": "^4.29.5"
  },
  "author": "",
  "license": "ISC"
}
```

Thanks in advance

**Comment by bahmutov** [comment]
I wonder if this startup time is due to transpiling and watching a lot of files for some reason

**Comment by indrajitbnikam** [comment]
@bahmutov is it because I configured it, so that I can use typescript? and which files are you talking about? I merely added any extra files apart from the name.spec.ts in integration folder

**Comment by dwelle** [comment]
On my win7 machine (everything running from SSD), from `npx cypress run` to first display of `(Run starting)` it takes:

- ~9s for `cypress-test-tiny` (1 empty spec)
- ~13s for a project of 18 specs of moderate sizes each

**Comment by indrajitbnikam** [comment]
Okay

**Comment by driscollAtBitGo** [comment]
I'm still seeing this, any progress made here? Discoveries on what the cause was? We're using both OSX and Alpine Linux builds and seeing the same behavior.

**Comment by pranav700** [comment]
Hello, facing the same issue, CLI takes atleast 10 seconds to start the cypress run. I'm using Cypress.json to pass the url still this happens. I am on windows. Any update please ? anyone..

**Comment by daniel78uk** [comment]
Same issue here I'm on OSX. Cypress 3.4.1

**Comment by Bkucera** [comment]
You might get a better idea of what's causing the slowdown by enabling debug logs.

set the env var `DEBUG=cypress*`

Windows: ```
set DEBUG=cypress*
npm run cypress:run
```
Unix: `DEBUG=cypress* npm run cypress:run`

**Comment by jpike88** [comment]
can someone from the cypress team like @jennifer-shehane please shed some light on why it takes that long? it adds up and blows out our test times by almost double the total time, and that's even on a lightning fast local development computer.

**Comment by StephRagazzi** [comment]
Same issue for me... It takes nearly 1 minute for tests to start on my machine ("waiting for localhost").
Using the Cypress debug feature, here the trace I got (only the operations that takes more than 1 sec):
```
cypress:browserify received user options: { browserifyOptions: { extensions: [ '.js', '.jsx', '.coffee' ], transform: [ [Function: transform], [Array], [Array] ], plugin: [] }, watchifyOptions: { ignoreWatch: [ '**/.git/**', '**/.nyc_output/**', '**/.sass-cache/**', '**/bower_components/**', '**/coverage/**', '**/node_modules/**' ] } } +13s

cypress:browserify finished bundling: C:\Users\me\AppData\Roaming\Cypress\cy\production\projects\myApp-255c3731241151d62bf2eff3120e6355\bundles\Specs\e2e\support\index.js +4s

cypress:server:controllers:spec sending spec { filePath: 'C:\\Users\\me\\AppData\\Roaming\\Cypress\\cy\\production\\projects\\myApp-255c3731241151d62bf2eff3120e6355\\bundles\\Specs\\e2e\\support\\index.js' } +49s

GET /__cypress/tests?p=Specs%5Ce2e%5Csupport%5Cindex.js-168 200 48882.590 ms - -
  cypress:server:events got request for event: get:project:status, { path: 'C:\\Dev\\env\\WebFrontEndApplications\\myApp' } +10s
  cypress:server:project get project status for client id undefined at path C:\Dev\env\WebFrontEndApplications\myApp +10s
  cypress:server:project no project id +1ms
  cypress:server:events sending ipc data { type: 'get:project:status', data: { id: 0.9356686012821931, data: { path: 'C:\\Dev\\env\\WebFrontEndApplications\\myApp', state: 'VALID' } } } +11ms
  cypress:browserify finished bundling: C:\Users\me\AppData\Roaming\Cypress\cy\production\projects\myApp-255c3731241151d62bf2eff3120e6355\bundles\Specs\e2e\specs\LogEntriesRetrieval.feature +12s
  cypress:server:plugins promise resolved for id 'inv2' with value C:\Users\me\AppData\Roaming\Cypress\cy\production\projects\myApp-255c3731241151d62bf2eff3120e6355\bundles\Specs\e2e\specs\LogEntriesRetrieval.feature +12s
  cypress:server:plugins promise resolved for id 'inv4' with value C:\Users\me\AppData\Roaming\Cypress\cy\production\projects\myApp-255c3731241151d62bf2eff3120e6355\bundles\Specs\e2e\specs\LogEntriesRetrieval.feature +1ms
  cypress:server:controllers:spec sending spec { filePath: 'C:\\Users\\me\\AppData\\Roaming\\Cypress\\cy\\production\\projects\\myApp-255c3731241151d62bf2eff3120e6355\\bundles\\Specs\\e2e\\specs\\LogEntriesRetrieval.feature' } +12s

cypress:server:server resolving visit { url: 'http://localhost:4201/', headers: { host: 'localhost:4201', connection: 'Upgrade', pragma: 'no-cache', 'cache-control': 'no-cache', 'user-agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.88 Safari/537.36', upgrade: 'websocket', origin: 'http://localhost:4201', 'sec-websocket-version': '13', 'accept-encoding': 'gzip, deflate, br', 'accept-language': 'en-US,en;q=0.9', 'sec-websocket-key': 'zpNyvOo+NzH/MdAB9cBsXQ==', 'sec-websocket-extensions': 'permessage-deflate; client_max_window_bits' }, options: { auth: null, failOnStatusCode: true, retryOnNetworkFailure: true, retryOnStatusCodeFailure: false, method: 'GET', body: null, headers: {}, timeout: 30000 } } +58s
  
cypress:server:request got cookies from browser { reqUrl: 'http://localhost:4201/', cookies: [] } +23s
  
cypress:network:agent addRequest called { isHttps: false, href: 'http://localhost:4201/' } +26s
  
cypress:network:connect beginning getAddress { hostname: 'localhost', port: '4201' } +26s
  
cypress:proxy:http Entering stage { stage: 'IncomingRequest' } +24s

cypress:proxy:http:request-middleware proxying request { req: { method: 'GET', proxiedUrl: 'http://localhost:4201/', headers: { host: 'localhost:4201', 'proxy-connection': 'keep-alive', 'upgrade-insecure-requests': '1', 'user-agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.88 Safari/537.36', accept: 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9', 'sec-fetch-site': 'same-origin', 'sec-fetch-mode': 'nested-navigate',
referer: 'http://localhost:4201/__/', 'accept-encoding': 'gzip, deflate, br', 'accept-language': 'en-US,en;q=0.9', cookie: '__cypress.initial=true' } } } +24s
```

**Comment by colin-studiobravo** [comment]
For my installation Cypress is now taking more than 80 seconds to start -- due to this I no longer use it as part of my workflow.

**Comment by jennifer-shehane** [comment]
We need a full project provided to reproduce this. The information initially given, with the test code below is not enough to reproduce the issue. 

```js
describe('My First Cypress Test', () => {
  it('Not much', () => {
    cy.visit('https://www.google.co.in/');
  });
});
```

Here is a video of the test run that was originally given as an example. This run time in the video is expected. 

![](http://g.recordit.co/LTKgumpGkT.gif)

## We need a reproducible example

We cannot begin looking at the long load time without a completely reproducible example since we never see this problem. 

This means there is very likely something specific to your project or setup that is slowing down the start time (webpack config, typescript config, some other file watching, the application under test, proxy configuration, some specific config in the configuration file, etc etc etc)

**Comment by indrajitbnikam** [comment]
@jennifer-shehane That's not necessarily the case I have also tried running cypresses default tests with minimal node project skeleton on my windows machine. It just takes a lot of time. plus I tried running it on a Linux VM on my Windows Machine and it's blazing fast.

So not able to pinpoint the cause since I tried running it in every scenario. Unfortunately, It always take significant time while running on windows machine no matter what.

**Comment by coder-shanks** [comment]
@jennifer-shehane Are you guys even working on this issue? This is really pain point for us as we wanted to use Cypress for our workflows. We are just waiting with NO response from your side

**Comment by dwelle** [comment]
Note that this may be related to the nature of Windows (semi micro-kernel architecture, NTFS filesystem etc.), and ultimately unsolvable (although, it can be possibly improved to a degree --- but no idea how).

**Comment by colin-studiobravo** [comment]
> For my installation Cypress is now taking more than 80 seconds to start -- due to this I no longer use it as part of my workflow.

I found this may have been an issue with a package used by `cypress-cucumber-preprocessor` -- after updating to version `2.0+` tests begin running in about 12 seconds.

**Comment by colin-studiobravo** [comment]
@StephRagazzi I notice you're also using Cucumber -- have you tried updating the preprocessor to version `2.0.1`?

**Comment by StephRagazzi** [comment]
> @StephRagazzi I notice you're also using Cucumber -- have you tried updating the preprocessor to version `2.0.1`?

Yes! same for me, after upgrading to the latest version of the Cypress Cucumber package, it runs after 15 sec!!

**Comment by Bkucera** [comment]
We're working on giving more logging during the preprocessor step. Currently we hide stderr, but we're going to change that, so you'll be able to see webpack output for example. https://github.com/cypress-io/cypress/issues/5765

**Comment by Jazzepi** [comment]
I had a problem like this. My startup times were around 40 seconds with not-very-many tests. Once the system started up it ran at a reasonable pace. I also had pretty bad stability issues where messing with the electron cypress window could cause the entire thing to crash.

The crux of the problem was that I was running Cypress itself (NOT THE TESTS) in a docker instance. This meant that the electron app Cypress was using my /dev/shm partition. Inside a docker this partition starts at 64 megabytes. My assumption is that the slowness to startup was because of Cypress-electron thrashing on that small space, being forced to use the real filesystem somehow (instead of in memory operations), and/or the smallness of the space causing some kind of exceptional problem where it fills up.

Once I mounted my external `/dev/shm` drive everything was 100% faster, and I haven't had a crash since then.

`--volume=/dev/shm:/dev/shm`

So if you're running Cypress in a Docker container make sure to increase your /dev/shm size by mounting it, or providing this command line option. No idea if 256 megabytes is enough, but this is the style of the command. `--shm-size=256m`

Perhaps Cypress could query the `/dev/shm` before it starts and print out an error or warning if the /dev/shm seems too small, or is the docker default.

**Comment by jennifer-shehane** [comment]
Unfortunately we have to close this issue as there is not enough information to reproduce the problem. This does not mean that your issue is not happening - it just means that we do not have a path to move forward. 

Please comment in this issue with a reproducible example ([or some information we previously asked for](https://github.com/cypress-io/cypress/issues/3692#issuecomment-573555888)) and we will consider reopening the issue.

**Comment by idvorkin** [comment]
I'm using cypress to run unit tests.  On my intel macbook pro 16", my unit tests take **124 ms**, but the cypress run call takes **11s**:
```
✔  All specs passed!                        124ms 
npx cypress run --config video=false --env coverage=false --spec '**/ut_*'  
   10.82s user 5.14s system 121% cpu 13.093 total
```

I have other things slowing down my cypress runs (typescript, nyc based coverage), but I captured  timings by setting debug flags,  and there seem to be several things taking high milliseconds. Curious if these are expected, or if there are flags that can be set to speed them. Hopefully this output helps: 

*(I don't know how many of these are double counting due to nested timing)*

```
  time taken detecting browsers (darwin workaround): 1.298s
  cypress:server:browsers browsers.kill called with no active instance +608ms
  cypress:server:browsers:electron debugger: enable Console and Network +955ms
  cypress:server:socket-base automation:request get:cookies { domain: 'localhost' } +2s
  cypress:proxy:http:util:buffers resetting buffers +3s
  cypress:server:browsers killing browser process +2s
  cypress:server:preprocessor removeFile ut_prompts.ts +2s
```

Full output of timings of 100ms and above: 

```
  cypress:cli:cli variable-length opts parsed { args: [ '/Users/idvorkin_home/homebrew/Cellar/node/17.0.1/bin/node', '/Users/idvorkin_home/gits/idvorkin.github.io/node_modules/.bin/cypress', 'run', '--config', 'video=false', '--env', 'coverage=false', '--spec', '**/ut_*' ], opts: Command { _events: [Object: null prototype] { 'option:browser': [Function (anonymous)], 'option:ci-build-id': [Function (anonymous)], 'option:config': [Function (anonymous)], 'option:config-file': [Function (anonymous)], 'option:env': [Function (anonymous)], 'option:group': [Function (anonymous)], 'option:key': [Function (anonymous)], 'option:headed': [Function (anonymous)], 'option:headless': [Function (anonymous)], 'option:no-exit': [Function (anonymous)], 'option:parallel': [Function (anonymous)], 'option:port': [Function (anonymous)], 'option:project': [Function (anonymous)], 'option:quiet': [Function (anonymous)], 'option:record': [Function (anonymous)], 'option:reporter': [Function (anonymous)], 'option:reporter-options': [Function (anonymous)], 'option:spec': [Function (anonymous)], 'option:tag': [Function (anonymous)], 'option:dev': [Function (anonymous)] }, _eventsCount: 20, _maxListeners: undefined, commands: [], options: [ [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option] ], parent: Command { _events: [Object: null prototype], _eventsCount: 1, _maxListeners: undefined, commands: [Array], options: [Array], parent: null, _allowUnknownOption: false, _args: [], rawArgs: [Array], _scriptPath: '/Users/idvorkin_home/gits/idvorkin.github.io/node_modules/.bin/cypress', _name: 'cypress', _optionValues: {}, _storeOptionsAsProperties: true, _passCommandToAction: true, _actionResults: [], _actionHandler: null, _executableHandler: false, _executableFile: null, _defaultCommandName: null, _exitCallback: null, _aliases: [], _hidden: false, _helpFlags: '-h, --help', _helpDescription: 'display help for command', _helpShortFlag: '-h', _helpLongFlag: '--help', _hasImplicitHelpCommand: undefined, _helpCommandName: 'help', _helpCommandnameAndArgs: 'help [command]', _helpCommandDescription: 'display help for command', _usage: '<command> [options]', args: [Array], [Symbol(kCapture)]: false }, _allowUnknownOption: false, _args: [], rawArgs: null, _scriptPath: null, _name: 'run', _optionValues: {}, _storeOptionsAsProperties: true, _passCommandToAction: true, _actionResults: [], _actionHandler: [Function: listener], _executableHandler: false, _executableFile: null, _defaultCommandName: null, _exitCallback: null, _aliases: [], _hidden: false, _helpFlags: '-h, --help', _helpDescription: 'display help for command', _helpShortFlag: '-h', _helpLongFlag: '--help', _hasImplicitHelpCommand: 0, _helpCommandName: 'help', _helpCommandnameAndArgs: 'help [command]', _helpCommandDescription: 'display help for command', _usage: '[options]', _description: 'Runs Cypress tests from the CLI without the GUI', _argsDescription: undefined, exit: true, config: 'video=false', env: 'coverage=false', spec: '**/ut_*', args: [], [Symbol(kCapture)]: false } } +190ms
  cypress:cli parsed cli options { config: 'video=false', env: 'coverage=false', spec: '**/ut_*' } +195ms cypress:cli verifying Cypress app +0ms
  cypress:server:appdata path: /Users/idvorkin_home/Library/Application Support/Cypress/cy/production +797ms
  cypress:server:appdata path: /Users/idvorkin_home/Library/Application Support/Cypress/cy/production/cache +997ms
  cypress:server:appdata path: /Users/idvorkin_home/Library/Application Support/Cypress/cy/production +1s
  cypress:server:cypress starting in mode run with options { _: [ '/Users/idvorkin_home/Library/Caches/Cypress/9.0.0/Cypress.app/Contents/MacOS/Cypress' ], sandbox: false, runProject: '/Users/idvorkin_home/gits/idvorkin.github.io', config: { video: false, env: { coverage: false } }, spec: [ '/Users/idvorkin_home/gits/idvorkin.github.io/**/ut_*' ], cwd: '/Users/idvorkin_home/gits/idvorkin.github.io', userNodePath: '/Users/idvorkin_home/homebrew/Cellar/node/17.0.1/bin/node', userNodeVersion: '17.0.1', invokedFromCli: true, testingType: 'e2e', projectRoot: '/Users/idvorkin_home/gits/idvorkin.github.io' } +998ms
time taken detecting browsers (darwin workaround): 1.298s
  cypress:server:browsers:utils found browsers { browsers: [ { name: 'chrome', family: 'chromium', channel: 'stable', displayName: 'Chrome', version: '96.0.4664.45', path: '/Applications/Google Chrome.app/Contents/MacOS/Google Chrome', minSupportedVersion: 64, majorVersion: 96 }, { name: 'edge', family: 'chromium', channel: 'stable', displayName: 'Edge', version: '95.0.1020.53', path: '/Applications/Microsoft Edge.app/Contents/MacOS/Microsoft Edge', minSupportedVersion: 79, majorVersion: 95 } ] } +1s
  cypress:server:run found all system browsers [ { name: 'chrome', family: 'chromium', channel: 'stable', displayName: 'Chrome', version: '96.0.4664.45', path: '/Applications/Google Chrome.app/Contents/MacOS/Google Chrome', minSupportedVersion: 64, majorVersion: 96 }, { name: 'edge', family: 'chromium', channel: 'stable', displayName: 'Edge', version: '95.0.1020.53', path: '/Applications/Microsoft Edge.app/Contents/MacOS/Microsoft Edge', minSupportedVersion: 79, majorVersion: 95 }, { name: 'electron', channel: 'stable', family: 'chromium', displayName: 'Electron', version: '94.0.4606.81', path: '', majorVersion: 94, info: 'Electron is the default browser that comes with Cypress. This is the default browser that runs in headless mode. Selecting this browser is useful when debugging. The version number indicates the underlying Chromium version that Electron uses.' } ] +1s
  cypress:server:plugins:child require pluginsFile +295ms
  cypress:server:plugins:child register event task with id 2 +162ms
  cypress:server:plugins resolving typescript with projectRoot '/Users/idvorkin_home/gits/idvorkin.github.io' +553ms
  cypress:server:plugins:child register event file:preprocessor with id 3 +322ms
  cypress:server:plugins register plugins process event _get:task:body with id 0 +2s
  cypress:server:project plugin config yielded: { animationDistanceThreshold: 5, baseUrl: null, blockHosts: null, browsers: [ { name: 'chrome', family: 'chromium', channel: 'stable', displayName: 'Chrome', version: '96.0.4664.45', path: '/Applications/Google Chrome.app/Contents/MacOS/Google Chrome', minSupportedVersion: 64, majorVersion: 96 }, { name: 'edge', family: 'chromium', channel: 'stable', displayName: 'Edge', version: '95.0.1020.53', path: '/Applications/Microsoft Edge.app/Contents/MacOS/Microsoft Edge', minSupportedVersion: 79, majorVersion: 95 }, { name: 'electron', channel: 'stable', family: 'chromium', displayName: 'Electron', version: '94.0.4606.81', path: '', majorVersion: 94, info: 'Electron is the default browser that comes with Cypress. This is the default browser that runs in headless mode. Selecting this browser is useful when debugging. The version number indicates the underlying Chromium version that Electron uses.' } ], chromeWebSecurity: true, clientCertificates: [], component: {}, componentFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/component', configFile: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress.json', defaultCommandTimeout: 4000, downloadsFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/downloads', e2e: {}, env: { coverage: false, codeCoverageTasksRegistered: true }, execTimeout: 60000, experimentalFetchPolyfill: false, experimentalInteractiveRunEvents: false, experimentalSessionSupport: false, experimentalSourceRewriting: false, experimentalStudio: false, fileServerFolder: '/Users/idvorkin_home/gits/idvorkin.github.io', fixturesFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/fixtures', hosts: null, ignoreTestFiles: '*.hot-update.js', includeShadowDom: false, integrationFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/integration', modifyObstructiveCode: true, numTestsKeptInMemory: 0, pageLoadTimeout: 60000, pluginsFile: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/plugins/index.js', port: null, projectId: null, projectRoot: '/Users/idvorkin_home/gits/idvorkin.github.io', redirectionLimit: 20, reporter: 'spec', reporterOptions: null, requestTimeout: 5000, resolvedNodePath: '/Users/idvorkin_home/homebrew/Cellar/node/17.0.1/bin/node', resolvedNodeVersion: '17.0.1', responseTimeout: 30000, retries: { runMode: 0, openMode: 0 }, screenshotOnRunFailure: true, screenshotsFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/screenshots', scrollBehavior: 'top', slowTestThreshold: 10000, supportFile: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/support/index.js', supportFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/support', taskTimeout: 60000, testFiles: '**/*.*', testingType: 'e2e', trashAssetsBeforeRuns: true, userAgent: null, version: '9.0.0', video: false, videoCompression: 32, videoUploadOnPasses: true, videosFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/videos', viewportHeight: 660, viewportWidth: 1000, waitForAnimations: true, watchForFileChanges: false } +2s
  cypress:server:config updateWithPluginValues { cfg: { projectRoot: '/Users/idvorkin_home/gits/idvorkin.github.io', projectName: 'idvorkin.github.io', rawJson: { envFile: {}, projectRoot: '/Users/idvorkin_home/gits/idvorkin.github.io', projectName: 'idvorkin.github.io' }, configFile: 'cypress.json', morgan: false, isTextTerminal: true, socketId: 'zrb04', report: true, browsers: [ [Object], [Object], [Object] ], video: false, animationDistanceThreshold: 5, baseUrl: null, blockHosts: null, chromeWebSecurity: true, clientCertificates: [], component: {}, componentFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/component', defaultCommandTimeout: 4000, downloadsFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/downloads', e2e: {}, env: { coverage: false }, execTimeout: 60000, experimentalFetchPolyfill: false, experimentalInteractiveRunEvents: false, experimentalSessionSupport: false, experimentalSourceRewriting: false, experimentalStudio: false, fileServerFolder: '/Users/idvorkin_home/gits/idvorkin.github.io', fixturesFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/fixtures', ignoreTestFiles: '*.hot-update.js', includeShadowDom: false, integrationFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/integration', modifyObstructiveCode: true, numTestsKeptInMemory: 0, pageLoadTimeout: 60000, pluginsFile: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/plugins/index.js', port: null, projectId: null, redirectionLimit: 20, reporter: 'spec', reporterOptions: null, requestTimeout: 5000, resolvedNodePath: '/Users/idvorkin_home/homebrew/Cellar/node/17.0.1/bin/node', resolvedNodeVersion: '17.0.1', responseTimeout: 30000, retries: { runMode: 0, openMode: 0 }, screenshotOnRunFailure: true, screenshotsFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/screenshots', slowTestThreshold: 10000, scrollBehavior: 'top', supportFile: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/support/index.js', supportFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/support', taskTimeout: 60000, testFiles: '**/*.*', trashAssetsBeforeRuns: true, userAgent: null, videoCompression: 32, videosFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/videos', videoUploadOnPasses: true, viewportHeight: 660, viewportWidth: 1000, waitForAnimations: true, watchForFileChanges: false, autoOpen: false, clientRoute: '/__/', devServerPublicPathRoute: '/__cypress/src', hosts: null, namespace: '__cypress', reporterRoute: '/__cypress/reporter', socketIoCookie: '__socket.io', socketIoRoute: '/__socket.io', xhrRoute: '/xhrs/', cypressEnv: 'production', resolved: { animationDistanceThreshold: [Object], baseUrl: [Object], blockHosts: [Object], chromeWebSecurity: [Object], clientCertificates: [Object], component: [Object], componentFolder: [Object], defaultCommandTimeout: [Object], downloadsFolder: [Object], e2e: [Object], env: [Object], execTimeout: [Object], experimentalFetchPolyfill: [Object], experimentalInteractiveRunEvents: [Object], experimentalSessionSupport: [Object], experimentalSourceRewriting: [Object], experimentalStudio: [Object], fileServerFolder: [Object], fixturesFolder: [Object], ignoreTestFiles: [Object], includeShadowDom: [Object], integrationFolder: [Object], modifyObstructiveCode: [Object], numTestsKeptInMemory: [Object], pageLoadTimeout: [Object], pluginsFile: [Object], port: [Object], projectId: [Object], redirectionLimit: [Object], reporter: [Object], reporterOptions: [Object], requestTimeout: [Object], resolvedNodePath: [Object], resolvedNodeVersion: [Object], responseTimeout: [Object], retries: [Object], screenshotOnRunFailure: [Object], screenshotsFolder: [Object], slowTestThreshold: [Object], scrollBehavior: [Object], supportFile: [Object], supportFolder: [Object], taskTimeout: [Object], testFiles: [Object], trashAssetsBeforeRuns: [Object], userAgent: [Object], video: [Object], videoCompression: [Object], videosFolder: [Object], videoUploadOnPasses: [Object], viewportHeight: [Object], viewportWidth: [Object], waitForAnimations: [Object], watchForFileChanges: [Object], browsers: [Object], hosts: [Object] }, parentTestsFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress', parentTestsFolderDisplay: 'idvorkin.github.io/cypress', scaffoldedFiles: [ [Object] ] }, overrides: { animationDistanceThreshold: 5, baseUrl: null, blockHosts: null, browsers: [ [Object], [Object], [Object] ], chromeWebSecurity: true, clientCertificates: [], component: {}, componentFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/component', configFile: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress.json', defaultCommandTimeout: 4000, downloadsFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/downloads', e2e: {}, env: { coverage: false, codeCoverageTasksRegistered: true }, execTimeout: 60000, experimentalFetchPolyfill: false, experimentalInteractiveRunEvents: false, experimentalSessionSupport: false, experimentalSourceRewriting: false, experimentalStudio: false, fileServerFolder: '/Users/idvorkin_home/gits/idvorkin.github.io', fixturesFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/fixtures', hosts: null, ignoreTestFiles: '*.hot-update.js', includeShadowDom: false, integrationFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/integration', modifyObstructiveCode: true, numTestsKeptInMemory: 0, pageLoadTimeout: 60000, pluginsFile: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/plugins/index.js', port: null, projectId: null, projectRoot: '/Users/idvorkin_home/gits/idvorkin.github.io', redirectionLimit: 20, reporter: 'spec', reporterOptions: null, requestTimeout: 5000, resolvedNodePath: '/Users/idvorkin_home/homebrew/Cellar/node/17.0.1/bin/node', resolvedNodeVersion: '17.0.1', responseTimeout: 30000, retries: { runMode: 0, openMode: 0 }, screenshotOnRunFailure: true, screenshotsFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/screenshots', scrollBehavior: 'top', slowTestThreshold: 10000, supportFile: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/support/index.js', supportFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/support', taskTimeout: 60000, testFiles: '**/*.*', testingType: 'e2e', trashAssetsBeforeRuns: true, userAgent: null, version: '9.0.0', video: false, videoCompression: 32, videoUploadOnPasses: true, videosFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/videos', viewportHeight: 660, viewportWidth: 1000, waitForAnimations: true, watchForFileChanges: false } } +2s
  cypress:config:validator validating configuration +2s
  cypress:server:validation browsers [ { name: 'chrome', family: 'chromium', channel: 'stable', displayName: 'Chrome', version: '96.0.4664.45', path: '/Applications/Google Chrome.app/Contents/MacOS/Google Chrome', minSupportedVersion: 64, majorVersion: 96 }, { name: 'edge', family: 'chromium', channel: 'stable', displayName: 'Edge', version: '95.0.1020.53', path: '/Applications/Microsoft Edge.app/Contents/MacOS/Microsoft Edge', minSupportedVersion: 79, majorVersion: 95 }, { name: 'electron', channel: 'stable', family: 'chromium', displayName: 'Electron', version: '94.0.4606.81', path: '', majorVersion: 94, info: 'Electron is the default browser that comes with Cypress. This is the default browser that runs in headless mode. Selecting this browser is useful when debugging. The version number indicates the underlying Chromium version that Electron uses.' } ] +2s
  cypress:server:appdata path: /Users/idvorkin_home/Library/Application Support/Cypress/cy/production/proxy +3s
  cypress:server:scaffold support folder /Users/idvorkin_home/gits/idvorkin.github.io/cypress/support, support file /Users/idvorkin_home/gits/idvorkin.github.io/cypress/support/index.js +2s
  cypress:server:settings resolved configObject {} +2s
  cypress:server:run project created and opened with config { configFile: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress.json', env: { codeCoverageTasksRegistered: true, coverage: false }, testingType: 'e2e', version: '9.0.0', projectRoot: '/Users/idvorkin_home/gits/idvorkin.github.io', projectName: 'idvorkin.github.io', rawJson: { envFile: {}, projectRoot: '/Users/idvorkin_home/gits/idvorkin.github.io', projectName: 'idvorkin.github.io' }, morgan: false, isTextTerminal: true, socketId: 'zrb04', report: true, browsers: [ { name: 'chrome', family: 'chromium', channel: 'stable', displayName: 'Chrome', version: '96.0.4664.45', path: '/Applications/Google Chrome.app/Contents/MacOS/Google Chrome', minSupportedVersion: 64, majorVersion: 96 }, { name: 'edge', family: 'chromium', channel: 'stable', displayName: 'Edge', version: '95.0.1020.53', path: '/Applications/Microsoft Edge.app/Contents/MacOS/Microsoft Edge', minSupportedVersion: 79, majorVersion: 95 }, { name: 'electron', channel: 'stable', family: 'chromium', displayName: 'Electron', version: '94.0.4606.81', path: '', majorVersion: 94, info: 'Electron is the default browser that comes with Cypress. This is the default browser that runs in headless mode. Selecting this browser is useful when debugging. The version number indicates the underlying Chromium version that Electron uses.' } ], video: false, animationDistanceThreshold: 5, baseUrl: null, blockHosts: null, chromeWebSecurity: true, clientCertificates: [], component: {}, componentFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/component', defaultCommandTimeout: 4000, downloadsFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/downloads', e2e: {}, execTimeout: 60000, experimentalFetchPolyfill: false, experimentalInteractiveRunEvents: false, experimentalSessionSupport: false, experimentalSourceRewriting: false, experimentalStudio: false, fileServerFolder: '/Users/idvorkin_home/gits/idvorkin.github.io', fixturesFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/fixtures', ignoreTestFiles: '*.hot-update.js', includeShadowDom: false, integrationFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/integration', modifyObstructiveCode: true, numTestsKeptInMemory: 0, pageLoadTimeout: 60000, pluginsFile: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/plugins/index.js', port: 57966, projectId: null, redirectionLimit: 20, reporter: 'spec', reporterOptions: null, requestTimeout: 5000, resolvedNodePath: '/Users/idvorkin_home/homebrew/Cellar/node/17.0.1/bin/node', resolvedNodeVersion: '17.0.1', responseTimeout: 30000, retries: { runMode: 0, openMode: 0 }, screenshotOnRunFailure: true, screenshotsFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/screenshots', slowTestThreshold: 10000, scrollBehavior: 'top', supportFile: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/support/index.js', supportFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/support', taskTimeout: 60000, testFiles: '**/*.*', trashAssetsBeforeRuns: true, userAgent: null, videoCompression: 32, videosFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/videos', videoUploadOnPasses: true, viewportHeight: 660, viewportWidth: 1000, waitForAnimations: true, watchForFileChanges: false, autoOpen: false, clientRoute: '/__/', devServerPublicPathRoute: '/__cypress/src', hosts: null, namespace: '__cypress', reporterRoute: '/__cypress/reporter', socketIoCookie: '__socket.io', socketIoRoute: '/__socket.io', xhrRoute: '/xhrs/', cypressEnv: 'production', resolved: { animationDistanceThreshold: { value: 5, from: 'default' }, baseUrl: { value: null, from: 'default' }, blockHosts: { value: null, from: 'default' }, chromeWebSecurity: { value: true, from: 'default' }, clientCertificates: { value: [], from: 'default' }, component: { value: {}, from: 'default' }, componentFolder: { value: 'cypress/component', from: 'default' }, defaultCommandTimeout: { value: 4000, from: 'default' }, downloadsFolder: { value: 'cypress/downloads', from: 'default' }, e2e: { value: {}, from: 'default' }, env: { coverage: [Object], codeCoverageTasksRegistered: [Object] }, execTimeout: { value: 60000, from: 'default' }, experimentalFetchPolyfill: { value: false, from: 'default' }, experimentalInteractiveRunEvents: { value: false, from: 'default' }, experimentalSessionSupport: { value: false, from: 'default' }, experimentalSourceRewriting: { value: false, from: 'default' }, experimentalStudio: { value: false, from: 'default' }, fileServerFolder: { value: '', from: 'default' }, fixturesFolder: { value: 'cypress/fixtures', from: 'default' }, ignoreTestFiles: { value: '*.hot-update.js', from: 'default' }, includeShadowDom: { value: false, from: 'default' }, integrationFolder: { value: 'cypress/integration', from: 'default' }, modifyObstructiveCode: { value: true, from: 'default' }, numTestsKeptInMemory: { value: 0, from: 'config' }, pageLoadTimeout: { value: 60000, from: 'default' }, pluginsFile: { value: 'cypress/plugins', from: 'default' }, port: { value: null, from: 'default' }, projectId: { value: null, from: 'default' }, redirectionLimit: { value: 20, from: 'default' }, reporter: { value: 'spec', from: 'default' }, reporterOptions: { value: null, from: 'default' }, requestTimeout: { value: 5000, from: 'default' }, resolvedNodePath: { value: null, from: 'default' }, resolvedNodeVersion: { value: null, from: 'default' }, responseTimeout: { value: 30000, from: 'default' }, retries: { value: [Object], from: 'default' }, screenshotOnRunFailure: { value: true, from: 'default' }, screenshotsFolder: { value: 'cypress/screenshots', from: 'default' }, slowTestThreshold: { value: 10000, from: 'default' }, scrollBehavior: { value: 'top', from: 'default' }, supportFile: { value: 'cypress/support', from: 'default' }, supportFolder: { value: false, from: 'default' }, taskTimeout: { value: 60000, from: 'default' }, testFiles: { value: '**/*.*', from: 'default' }, trashAssetsBeforeRuns: { value: true, from: 'default' }, userAgent: { value: null, from: 'default' }, video: { value: false, from: 'cli' }, videoCompression: { value: 32, from: 'default' }, videosFolder: { value: 'cypress/videos', from: 'default' }, videoUploadOnPasses: { value: true, from: 'default' }, viewportHeight: { value: 660, from: 'default' }, viewportWidth: { value: 1000, from: 'default' }, waitForAnimations: { value: true, from: 'default' }, watchForFileChanges: { value: false, from: 'config' }, browsers: { value: [Array], from: 'default' }, hosts: { value: null, from: 'default' }, configFile: { value: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress.json', from: 'plugin' }, testingType: { value: 'e2e', from: 'plugin' }, version: { value: '9.0.0', from: 'plugin' } }, parentTestsFolder: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress', parentTestsFolderDisplay: 'idvorkin.github.io/cypress', scaffoldedFiles: [ { name: 'cypress', children: [Array] } ], proxyUrl: 'http://localhost:57966', browserUrl: 'http://localhost:57966/__/', reporterUrl: 'http://localhost:57966/__cypress/reporter', xhrUrl: '__cypress/xhrs/', proxyServer: 'http://localhost:57966', state: {} } +2s
  cypress:server:browsers searching for browser { nameOrPath: 'electron', filter: { name: 'electron', channel: 'stable' }, knownBrowsers: [ { name: 'chrome', family: 'chromium', channel: 'stable', displayName: 'Chrome', version: '96.0.4664.45', path: '/Applications/Google Chrome.app/Contents/MacOS/Google Chrome', minSupportedVersion: 64, majorVersion: 96 }, { name: 'edge', family: 'chromium', channel: 'stable', displayName: 'Edge', version: '95.0.1020.53', path: '/Applications/Microsoft Edge.app/Contents/MacOS/Microsoft Edge', minSupportedVersion: 79, majorVersion: 95 }, { name: 'electron', channel: 'stable', family: 'chromium', displayName: 'Electron', version: '94.0.4606.81', path: '', majorVersion: 94, info: 'Electron is the default browser that comes with Cypress. This is the default browser that runs in headless mode. Selecting this browser is useful when debugging. The version number indicates the underlying Chromium version that Electron uses.' } ] } +3s
  cypress:server:profilecleaner found 0 root level profile matches: [] +488ms
  cypress:server:specs ] +491ms
  cypress:server:run found '1' specs using spec pattern '[ '/Users/idvorkin_home/gits/idvorkin.github.io/**/ut_*' ]': [ 'unit/ut_prompts.ts' ] +500ms
  cypress:server:run formatting Node version. { version: '17.0.1', path: '/Users/idvorkin_home/homebrew/Cellar/node/17.0.1/bin/node' } +101ms
  cypress:server:plugins plugin event registered? { event: 'before:run', isRegistered: false } +696ms
  cypress:server:open_project resetting project state, preparing to launch browser electron for spec { name: 'unit/ut_prompts.ts', relative: 'cypress/integration/unit/ut_prompts.ts', absolute: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/integration/unit/ut_prompts.ts', specType: 'integration' } options { width: 1280, height: 720, show: false, onCrashed: [Function: onCrashed], onNewWindow: [Function: onNewWindow], automationMiddleware: { onBeforeRequest: [Function: onBeforeRequest], onAfterResponse: [Function: onAfterResponse] }, projectRoot: '/Users/idvorkin_home/gits/idvorkin.github.io', onWarning: [Function (anonymous)] } +2s
  cypress:server:project resetting project instance /Users/idvorkin_home/gits/idvorkin.github.io +611ms
  cypress:server:server-base Setting remoteAuth undefined +617ms
  cypress:server:browsers browsers.kill called with no active instance +608ms
  cypress:server:saved_state noop saved state +645ms
  cypress:server:server-base Getting remote state: { auth: undefined, props: null, origin: 'http://localhost:57966', strategy: 'file', visiting: undefined, domainName: 'localhost', fileServer: 'http://localhost:57967' } +166ms
  cypress:server:server-base Got CONNECT request from localhost:57966 +867ms
  cypress:https-proxy Writing browserSocket connection headers { url: 'localhost:57966', headLength: 0, headers: { host: 'localhost:57966', 'proxy-connection': 'keep-alive', 'user-agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Cypress/9.0.0 Chrome/94.0.4606.81 Electron/15.2.0 Safari/537.36' } } +2s
  cypress:server:browsers:electron debugger: enable Console and Network +955ms
  cypress:server:browsers browser opened +1s
  cypress:server:project socket:connected +1s
  cypress:server:run got socket connection { id: 'zrb04' } +1s
  cypress:server:server-base Getting remote state: { auth: undefined, props: null, origin: 'http://localhost:57966', strategy: 'file', visiting: undefined, domainName: 'localhost', fileServer: 'http://localhost:57967' } +130ms
  cypress:server:appdata path: /Users/idvorkin_home/Library/Application Support/Cypress/cy/production/projects/idvorkin.github.io-0b07ac8a731c76490881d1f2ee258931/bundles/cypress/integration/unit/ut_prompts.ts +2s
  cypress:server:plugins execute plugin event 'file:preprocessor' Node 'v16.5.0' with args: EventEmitter { _events: [Object: null prototype] { rerun: [Function (anonymous)] }, _eventsCount: 1, _maxListeners: undefined, filePath: '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/integration/unit/ut_prompts.ts', shouldWatch: false, outputPath: '/Users/idvorkin_home/Library/Application Support/Cypress/cy/production/projects/idvorkin.github.io-0b07ac8a731c76490881d1f2ee258931/bundles/cypress/integration/unit/ut_prompts.ts', [Symbol(kCapture)]: false } undefined undefined +1s
  cypress:server:plugins:child execute plugin event: file:preprocessor ({ eventId: 3, invocationId: 'inv1' }) +2s
  cypress-verbose:server:browsers:cdp_automation received networkRequestWillBeSent { documentURL: 'http://localhost:57966/__cypress/iframes/integration/unit/ut_prompts.ts', frameId: '63F3875313DCDC51D82744411B92B9FF', hasUserGesture: false, initiator: { stack: { callFrames: [Array] }, type: 'script' }, loaderId: 'A750E5041DDF368F3F0A9512DA2536A6', request: { headers: { Referer: 'http://localhost:57966/__cypress/iframes/integration/unit/ut_prompts.ts', 'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Cypress/9.0.0 Chrome/94.0.4606.81 Electron/15.2.0 Safari/537.36' }, initialPriority: 'High', isSameSite: true, method: 'GET', mixedContentType: 'none', referrerPolicy: 'strict-origin-when-cross-origin', url: 'http://localhost:57966/__cypress/tests?p=cypress/support/index.js' }, requestId: '61401.50', timestamp: 73734.076752, type: 'XHR', wallTime: 1637633934.892147 } +117ms
  cypress-verbose:proxy:http:util:prerequests queuing pre-request to be matched later { requestId: '61401.50', method: 'GET', url: 'http://localhost:57966/__cypress/tests?p=cypress/support/index.js', headers: { Referer: 'http://localhost:57966/__cypress/iframes/integration/unit/ut_prompts.ts', 'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Cypress/9.0.0 Chrome/94.0.4606.81 Electron/15.2.0 Safari/537.36' }, resourceType: 'xhr', originalResourceType: 'XHR' } [ { requestId: 'A750E5041DDF368F3F0A9512DA2536A6', method: 'GET', url: 'http://localhost:57966/__cypress/iframes/integration/unit/ut_prompts.ts', headers: { Referer: 'http://localhost:57966/__/', 'Upgrade-Insecure-Requests': '1', 'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Cypress/9.0.0 Chrome/94.0.4606.81 Electron/15.2.0 Safari/537.36' }, resourceType: 'other', originalResourceType: 'Document' } ] +117ms
  cypress:server:plugins:child execute plugin event: file:preprocessor ({ eventId: 3, invocationId: 'inv2' }) +167ms
  cypress:webpack user options: { typescript: '/Users/idvorkin_home/gits/idvorkin.github.io/node_modules/typescript/lib/typescript.js', webpackOptions: { mode: 'development', node: { global: true, __filename: true, __dirname: true }, module: { rules: [Array] }, resolve: { extensions: [Array], alias: [Object], plugins: [Array] }, entry: [ '/Users/idvorkin_home/gits/idvorkin.github.io/cypress/integration/unit/ut_prompts.ts' ], output: { path: '/Users/idvorkin_home/Library/Application Support/Cypress/cy/production/projects/idvorkin.github.io-0b07ac8a731c76490881d1f2ee258931/bundles/cypress/integration/unit', filename: 'ut_prompts.ts.js' }, devtool: 'inline-source-map' }, __typescriptSupportAdded: true } +140ms
  cypress:webpack typescript.createProgram() already overridden +143ms
  cypress:webpack typescript unmodified createProgram options { outDir: '/Users/idvorkin_home/gits/idvorkin.github.io/assets/js', allowJs: true, target: 6, sourceMap: true, inlineSourceMap: true, inlineSources: true, downlevelIteration: true, configFilePath: '/Users/idvorkin_home/gits/idvorkin.github.io/tsconfig.json', skipLibCheck: true, suppressOutputPathCheck: true } +233ms
  cypress:webpack finished bundling /Users/idvorkin_home/Library/Application Support/Cypress/cy/production/projects/idvorkin.github.io-0b07ac8a731c76490881d1f2ee258931/bundles/cypress/integration/unit/ut_prompts.ts.js +1s
Time: 1499ms
  cypress:server:controllers:spec sending spec { filePath: '/Users/idvorkin_home/Library/Application Support/Cypress/cy/production/projects/idvorkin.github.io-0b07ac8a731c76490881d1f2ee258931/bundles/cypress/integration/unit/ut_prompts.ts.js' } +2s
Time: 1544ms
  cypress:server:browsers:electron console message: { column: 11, level: 'log', line: 485, source: 'console-api', text: 'Skipping code coverage hooks', url: 'http://localhost:57966/__cypress/tests?p=cypress/support/index.js' } +2s
  cypress:server:project received runnables { id: 'r1', title: '', root: true, type: 'suite', file: 'cypress/integration/unit/ut_prompts.ts', retries: -1, _slow: 10000, hooks: [], tests: [], suites: [ { id: 'r2', title: 'Things I enjoy tests', root: false, type: 'suite', file: null, invocationDetails: [Object], retries: -1, _slow: 10000, hooks: [], tests: [Array], suites: [] }, { id: 'r4', title: 'Page Navigation works', root: false, type: 'suite', file: null, invocationDetails: [Object], retries: -1, _slow: 10000, hooks: [], tests: [Array], suites: [] }, { id: 'r7', title: 'Tree Walker', root: false, type: 'suite', file: null, invocationDetails: [Object], retries: -1, _slow: 10000, hooks: [], tests: [Array], suites: [] } ], runtimeConfig: {} } +2s
  cypress:server:reporter trying to load reporter: spec +4s
  cypress-verbose:server:browsers:cdp_automation received networkRequestWillBeSent { documentURL: 'http://localhost:57966/__/', frameId: '664F25B7BDC13E6E69B961F3703DB47A', hasUserGesture: false, initiator: { type: 'parser', url: 'http://localhost:57966/__cypress/runner/cypress_runner.css' }, loaderId: 'AE8A9C9F424B64FD35B46F5AE43F0C3D', request: { headers: { Origin: 'http://localhost:57966', Referer: 'http://localhost:57966/__cypress/runner/cypress_runner.css', 'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Cypress/9.0.0 Chrome/94.0.4606.81 Electron/15.2.0 Safari/537.36' }, initialPriority: 'VeryHigh', isSameSite: true, method: 'GET', mixedContentType: 'none', referrerPolicy: 'strict-origin-when-cross-origin', url: 'http://localhost:57966/__cypress/runner/fonts/fa-regular-400.woff2' }, requestId: '61401.38', timestamp: 73735.794772, type: 'Font', wallTime: 1637633936.610152 } +2s
  cypress-verbose:proxy:http:util:prerequests queuing pre-request to be matched later { requestId: '61401.38', method: 'GET', url: 'http://localhost:57966/__cypress/runner/fonts/fa-regular-400.woff2', headers: { Origin: 'http://localhost:57966', Referer: 'http://localhost:57966/__cypress/runner/cypress_runner.css', 'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Cypress/9.0.0 Chrome/94.0.4606.81 Electron/15.2.0 Safari/537.36' }, resourceType: 'font', originalResourceType: 'Font' } [ { requestId: 'A750E5041DDF368F3F0A9512DA2536A6', method: 'GET', url: 'http://localhost:57966/__cypress/iframes/integration/unit/ut_prompts.ts', headers: { Referer: 'http://localhost:57966/__/', 'Upgrade-Insecure-Requests': '1', 'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Cypress/9.0.0 Chrome/94.0.4606.81 Electron/15.2.0 Safari/537.36' }, resourceType: 'other', originalResourceType: 'Document' }, { requestId: '61401.50', method: 'GET', url: 'http://localhost:57966/__cypress/tests?p=cypress/support/index.js', headers: { Referer: 'http://localhost:57966/__cypress/iframes/integration/unit/ut_prompts.ts', 'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Cypress/9.0.0 Chrome/94.0.4606.81 Electron/15.2.0 Safari/537.36' }, resourceType: 'xhr', originalResourceType: 'XHR' }, { requestId: '61401.51', method: 'GET', url: 'http://localhost:57966/__cypress/tests?p=cypress/integration/unit/ut_prompts.ts', headers: { Referer: 'http://localhost:57966/__cypress/iframes/integration/unit/ut_prompts.ts', 'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Cypress/9.0.0 Chrome/94.0.4606.81 Electron/15.2.0 Safari/537.36' }, resourceType: 'xhr', originalResourceType: 'XHR' } ] +2s
  cypress:server:socket-base automation:request get:cookies { domain: 'localhost' } +2s
  cypress:proxy:http:util:buffers resetting buffers +3s
  6 passing (122ms)
  cypress:server:plugins plugin event registered? { event: 'after:spec', isRegistered: false } +2s
  cypress:server:run attempting to close the browser +2s
  cypress:server:browsers killing browser process +2s
  cypress:server:preprocessor removeFile /Users/idvorkin_home/gits/idvorkin.github.io/cypress/integration/unit/ut_prompts.ts +2s
  cypress:webpack close /Users/idvorkin_home/gits/idvorkin.github.io/cypress/integration/unit/ut_prompts.ts +232ms
  │ ✔  unit/ut_prompts.ts                       124ms        6        6        -        -        - │
    ✔  All specs passed!                        124ms        6        6        -        -        -
  cypress:server:cypress about to exit with code 0 +7s
  cypress:cli child event fired { event: 'exit', code: 0, signal: null } +12s
npx cypress run --config video=false --env coverage=false --spec '**/ut_*'  10.82s user 5.14s system 121% cpu 13.093 total
```

My full [project and configuration](https://github.com/idvorkin/idvorkin.github.io)

**Comment by vincerubinetti** [comment]
FWIW:

I have a `vue-cli` project with `cypress` preconfigured to run e2e tests, and it was taking [an ungodly amount of time](https://github.com/monarch-initiative/monarch-ui-new/runs/4388280230?check_suite_focus=true) (> 5 minutes) and causing GitHub Actions to timeout and cancel the workflow. Turning on cypress debug mode revealed that it was the capturing of video that was taking up most of the time. Setting `video: false` in cypress made it [run more reasonably](https://github.com/monarch-initiative/monarch-ui-new/runs/4388355342?check_suite_focus=true) (< 1m).

**Comment by debasisj** [comment]
hi guys, i understand this is closed but this is definitely an issue - we are now adding up to 30-40% extra time to our regression suite for specs just to wait for tests to be loaded - what is the root cause and is there a best practice to resolve it?

the more specs/tests we are adding, it getting slower. all of our specs are under `cypress/integration`

**Comment by trainoasis** [comment]
our tests in Github's CI now after migrating k8s cluster sometimes take 20 minutes 😱 Just one test sometimes takes 2-3 minutes by itself .. not sure what's happening

**Comment by MaksuCode** [comment]
I have the same problem with v13.12.0. 
It takes around 10-15s to load the test

**Comment by jennifer-shehane** [comment]
Please open a new issue with a reproducible example if you are facing this issue.

**Comment by MaksuCode** [comment]
@jennifer-shehane I created [this ](https://github.com/cypress-io/cypress/issues/29737) issue. I guess it might be related to faker library I use to generate test data before each test.
